### DISCUSSION TITLE
在嵌套工程中，如何取得 xmake 命令的调用目录

### DISCUSSION THREAD

**Discussion by afoolsbag** [xmake-io/xmake#2249] [Q&A] [answered]
## 情景

我有一个用于存放 C/C++ 备忘的代码仓库，里面搜集了我用过的所有库的备忘，按照库名作为文件夹分层，譬如：

```
MyMemo/
    BoostExample/
        ......
        CMakeLists.txt
        xmake.lua
    DoctestExample/
        ......
        CMakeLists.txt
        xmake.lua
    QtExample/
        ......
        CMakeLists.txt
        xmake.lua
    CMakeLists.txt
    xmake.lua
    ............
```

以前我使用 CMake 的 OPTION 和 GUI 来配合管理这个项目，用选项来开启编译哪个库，现在我想转用 xmake。

## 目的

xmake 的命令行真是十分好用，所以我想通过判断当前 xmake 的运行路径，来单独开启对某个子项目的编译：

譬如：

*   在 `MyMemo` 运行 xmake，通过获取运行路径，可以控制在根 `xmake.lua` 中包含所有子项目；
*   在 `MyMemo/QtExample` 运行 xmake，通过获取运行路径，可以控制在根 `xmake.lua` 中仅包含 `QtExample/xmake.lua` 子项目。

## 困难

其它方面我都可以自行编写脚本解决，但是**如何获取 xmake 的调用目录**？

我尝试过使用 `os.curdir`、`os.scriptdir` 和 `os.projectdir` 它们都无法满足我的要求。

我是 xmake 新手，任何建议都是有益，谢谢！

**Comment by waruqi** [comment, marked as answer]
搞这么麻烦干嘛，直接 includes 引入所有子目录 xmake.lua ，每个目标程序用 target 定义，不想默认编译的 set_default(false) 关了

xmake 编译默认的 targets
xmake target_name  编译指定的 target

**Reply by afoolsbag** [reply]
`add_requires` 有办法和对应的 `target` 同步开启关闭吗，各子项目的依赖有些多😂

**Reply by waruqi** [reply]
add_requires 是全局的，add_packages 跟 target 进行绑定
